src/v/cluster/tm_stm.cc

```diff
@@ -245,6 +317,53 @@ tm_stm::do_sync(model::timeout_clock::duration timeout) {
     co_return _insync_term;
 }
 
+ss::future<tm_stm::op_status> tm_stm::update_hosted_transactions(
+  model::term_id term, tm_tx_hosted_transactions hr) {
+    co_return co_await ss::with_gate(_gate, [this, term, hr = std::move(hr)] {
```

Comment:
use an RAII gate holder since this is a coroutine instead of ss::with_gate

Reply:
do you mean?

```
auto gh = _gate.hold();
co_return co_await do_update_hosted_transactions(term, std::move(hr));
```

Reply:
yes, exactly.